### PR TITLE
Standardize scalesWith across character UI

### PR DESCRIPTION
**Pull request by frzyc**
## Describe your changes

Hoist up a "scales with" list into a context to be used across Character UI (TC opt, opt, overview etc)
Add a "Scales with" alert to the optimization page
Remove the build constraint in TC builds, opting to use the build constraints from optimization page.

## Issue or discord link

- 

## Testing/validation
Screenshot 2024-09-25 164149

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Comment by nguyentvan7** [comment]
If I toggle spread on and off, it changes the order of the stats. Should we deterministically sort this to avoid the implication that one stat is weighted more? I might be overthinking it

image
image

**Comment by nguyentvan7** [comment]
Technically ER should be here, as I only enabled EOSF, and in general EOSF does just make burst scale off ER, but i guess not really known unless the character is currently equipping EOSF

image

e.g. when char has EOSF equipped, ER shows up
image

File: libs/gi/page-team/src/CharacterDisplay/Tabs/TabOptimize/ScalesWith.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { Alert } from '@mui/material'
+import { useContext } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+
+export default function ScalesWith() {
+  const { t } = useTranslation('page_character_optimize')
+  const { scalesWith } = useContext(OptTargetContext)
+  return (
+    !!scalesWith.size && (
+      <Alert severity="info" variant="filled">
```
Comment (nguyentvan7) [inline libs/gi/page-team/src/CharacterDisplay/Tabs/TabOptimize/ScalesWith.tsx:16]: Can we use the new standard for alert?
image

Comment (frzyc) [reply libs/gi/page-team/src/CharacterDisplay/Tabs/TabOptimize/ScalesWith.tsx:16]: you meant the `variant="standard"`? I'll give it a try.

**Comment by frzyc** [comment]
> Technically ER should be here, as I only enabled EOSF, and in general EOSF does just make burst scale off ER, but i guess not really known unless the character is currently equipping EOSF
> 
> image
> 
> e.g. when char has EOSF equipped, ER shows up image

Yeah that is an issue that would be pretty difficult to address, Another cornercase would be that if you currently have ESOF equipped, and it generates a build without ESOF, it would stills show it scaling with ER. I am honestly not sure if I want to commit the time to solve this since any solution I can think of involves a lot of spaghetti.

**Comment by frzyc** [comment]
> If I toggle spread on and off, it changes the order of the stats. Should we deterministically sort this to avoid the implication that one stat is weighted more? I might be overthinking it
> 
> image image

Random order would actually avoid the implication that there is a priority, no? Also yeah its not ranked (because how can we determine that even) so its not really an issue worth addressing.

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**